Fix normative language in notes (#1010)
unicode-org/message-format-wg: spec/intro.md

All other _identifiers_ in these categories are reserved for the use of implementations or users.

> [!IMPORTANT]
> Implementation-defined or user-defined _functions_ and _function_ _options_
> SHOULD use a _namespace_ as part of their _identifiers_

> > to avoid collisions with other implementations.

Reviewer: ```suggestion
> to help avoid collisions with other implementations.
```

All other _identifiers_ in these categories are reserved for the use of implementations or users.

> [!IMPORTANT]

> > Users defining custom _identifiers_ SHOULD use a _namespace_

Reviewer 2: It is not clear exactly what 'custom identifiers' means. Does that mean "all other identifiers"? If so, the text should say so

What does "users" mean in this context? Authors of messages? Authors of functions?

identifiers are used in functions, options, option values, and attributes.

The above implies that I (a message writter) should use a namespace in an option value that is a free-form string wit non-ascii. So instead of:

... name=René ...

I should write: name=med:René ...

Moreover, "to avoid collisions with other implementations." needs to be toned down, to "to help avoid collisions with other implementations.". Using a namespace is no guarantee of avoidance: there could easily be another 'med' namespace.

Author: Note that identifiers are not used in option values.

The point about attributes is valid; this needs to be fixed for that. If the following addresses your concerns, please merge this suggestion?

```suggestion
> Users defining custom _identifiers_ for _functions_ and _function_ _options_ SHOULD use a _namespace_
```

Reviewer: A better overall phrasing would be more like:

```suggestion
Implementation-defined or user-defined _functions_ and _function_ _options_ SHOULD use a _namespace_
as part of their _identifiers_
```

Author: Merging with editorial edits.
